### PR TITLE
Fix #117 and #118

### LINKED ISSUE #117: Use default tap reporter
Use tap-bark as default tap reporter because the output could be more concise than it is and also more user friendly

### PR DESCRIPTION
**Pull request by Jameskmonger**
# Description

Fix issues #117 and #118 

# Checklist

- [x] I am an awesome developer and proud of my code
- [x] I added / updated / removed relevant unit or integration tests to prove my change works
- [x] I ran all tests using ```npm test``` to make sure everything else still works
- [x] I ran ```npm run review``` to ensure the code adheres to the repository standards

# Additional Information

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #NPM
 node_modules
-npm-debug.log
+npm-debug.log*
```
Comment (jamesrichford) [inline .gitignore]: What's the reason for this change?

File: README.md [hunk 2]
```diff
@@ -46,6 +46,14 @@ alsatian --timeout 5000
 alsatian -t 5000
 ```
 
+#### TAP
+
+The default reporter is the super snazzy [tap-bark](http://github.com/alsatian-test/tap-bark). If you want raw TAP output (so that it can be piped into a reporter of your choice), you can use the `tap` argument.
+
+```
+alsatian --tap
```
Comment (jamesrichford) [inline README.md]: add a shorthand for tap output?

File: cli/alsatian-cli.ts [hunk 3]
```diff
@@ -1,18 +1,37 @@
 #! /usr/bin/env node
 
-import { createTestSet, TestRunner } from "../core/alsatian-core";
+import { createTestSet, TestRunner, TestOutput } from "../core/alsatian-core";
 import { CliTestRunner } from "./cli-test-runner";
 import { AlsatianCliOptions } from "./alsatian-cli-options";
 
+const {
```
Comment (jamesrichford) [inline cli/alsatian-cli.ts]: any reason not to use es6 import?

File: cli/alsatian-cli.ts [hunk 4]
```diff
@@ -1,18 +1,37 @@
...
 // get all arguments from the user
 let userArguments = new AlsatianCliOptions(process.argv.slice(2));
 
 // create test set from given file globs
 let testSet = createTestSet();
 testSet.addTestsFromFiles(userArguments.fileGlobs);
 
+let reporter: any;
+
+if (userArguments.tap) {
```
Comment (jamesrichford) [inline cli/alsatian-cli.ts]: test coverage has gone down :( looks like here is the issue - we have no tests here yet so may need a bit of work to write some tests around here. Will need them pre beta though so ideal to do them if possible. If too difficult no worries I'll do them later

File: package.json [hunk 5]
```diff
@@ -30,7 +30,7 @@
     "test": "npm run unit-tests",
     "test-dot": "node ./cli/alsatian-cli.js ./test/**/*.spec.js | tap-dot",
     "test-spec": "node ./cli/alsatian-cli.js ./test/**/*.spec.js | tap-spec",
-    "unit-tests": "istanbul cover ./cli/alsatian-cli.js \"./test/unit-tests/**/*.spec.js\" && npm run remap-coverage",
```
Comment (jamesrichford) [inline package.json]: has istanbul been removed intentionally?

**Comment by jamesrichford** [comment]
Hey @Jameskmonger, looking good - just tried the output in cmd and powershell and these seem a bit cooky (fragments being left / moving onto new lines) but also missing the percentage progress, totals at the end and errors being listed. Is this windows issue or missing from tap-bark?

File: .gitignore (hunk 1 above)
Comment (Jameskmonger) [inline .gitignore]: @jamesrichford `npm-debug` files are often saved with a timestamp on the file type (don't ask me why). e.g. `npm-debug.log12312238141`. At the moment they're not ignored from git

File: README.md (hunk 2 above)
Comment (Jameskmonger) [inline README.md]: @jamesrichford Could do. `-t` is taken, though ;)

File: cli/alsatian-cli.ts (hunk 3 above)
Comment (Jameskmonger) [inline cli/alsatian-cli.ts]: @jamesrichford No typings for `tap-bark` at the moment so it kicked off a bit. I can give it another go if you like.

File: package.json (hunk 5 above)
Comment (Jameskmonger) [inline package.json]: @jamesrichford Nope! I was testing and must have accidentally committed.

File: cli/alsatian-cli.ts (hunk 4 above)
Comment (Jameskmonger) [inline cli/alsatian-cli.ts]: @jamesrichford I'll have a look into it and see if there's anything I can do

**Comment by Jameskmonger** [comment]
I'll get on these @jamesrichford!

File: README.md (hunk 2 above)
Comment (jamesrichford) [inline README.md]: -T it is ;)

File: .gitignore (hunk 1 above)
Comment (jamesrichford) [inline .gitignore]: ah weird, maybe a GIT setting? That's fine then :)

File: cli/alsatian-cli.ts (hunk 3 above)
Comment (jamesrichford) [inline cli/alsatian-cli.ts]: Yeah, if we could get the typings working that would be legendary mate

**Comment by jamesrichford** [comment]
Nice one!!

**Comment by Jameskmonger** [comment]
I've no idea why `tap-parser` isn't emitting `complete` properly. It works for `tap-bark` alone but when it's implemented here it's never emitted. Mind taking a look?

**Comment by jamesrichford** [comment]
Without a computer and just looking at the code my first guess would be that you are only piping stdout to tap bark so it can get that info but nothing else. To determine whether the process has ended it'll need to do something along the lines of ```process.on("close", (code) => handleIt(code));``` but it doesn't have access to the process. May need to pass the process on I'd suspect

**Comment by jamesrichford** [comment]
Accepting change in coverage temporarily (to be addressed in #54
